### DISCUSSION TITLE
Support glibc-hwcaps in rpm

### DISCUSSION THREAD

**Discussion by romulasry** [rpm-software-management/rpm#2022] [Ideas]
"dancermak
    5 days ago by dancermak | Reply

    This sounds very intriguing! I have a few notes about this:
        you might be interested in this (sadly stalled) upstream PR:
https://github.com/rpm-software-management/rpm/pull/1035 which adds better detection of
the currently running microarchitecture
        once rpm gains the ability to automatically generate subpackages
(https://github.com/rpm-software-management/rpm/pull/1485), this could be completely
automated"
https://lists.fedorahosted.org/archives/list/devel@lists.fedoraproject.org/message/5GAPNLIJGNCBZCP2L2CDY6I6TZLZD6EB/

Also see: x86_64-v2 in Fedora
https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/5EJ7T27EBBSJKMDY2MKCKR5ZIRBCKPJ6/

**Comment by pmatilai** [comment]
The more relevant link would seem to be https://hackweek.opensuse.org/all/projects/support-glibc-hwcaps-and-micro-architecture-package-generation

**Comment by davidlt** [comment]
Note that RISC-V plans to have profiles too (see: https://github.com/riscv/riscv-profiles/blob/main/profiles.adoc ). So right now Linux distributions target RV64GC (which expands to a longer ISA string). That is encapsulated into RVA20 profile. There should be another two RVA22 and RVA23 a bit later. The idea is that RVA20 and RVA23 should "major" profiles and RVA22 a "minor" one. If distributions don't want to pick up all of the profiles, they could focus on "major" ones. Think that RVA20 -> RVA23 is like ARMv8 -> ARMv9.

**Comment by romulasry** [comment]
https://github.com/rpm-software-management/rpm/pull/2315
